src/MSBuild/TerminalLogger/TerminalLogger.cs

```diff
@@ -28,7 +28,7 @@ internal sealed class TerminalLogger : INodeLogger
 {
     private const string FilePathPattern = " -> ";
     private const char PatternSeparator = '|';
-    private readonly string _immediateMessagePattern = $@"\[CredentialProvider\]{PatternSeparator}--interactive";
+    private readonly Regex _immediateMessageRegex = new Regex($@"\[CredentialProvider\]{PatternSeparator}--interactive");
```

Comment:
Consider doing the full source-generator thing for .NET 8. Example in #9411.

Reply:
done